xarray/core/groupby.py

```diff
@@ -676,6 +682,78 @@ def sizes(self) -> Mapping[Hashable, int]:
             self._sizes = self._obj.isel({self._group_dim: index}).sizes
         return self._sizes
 
+    def shuffle(self, chunks: T_Chunks = None):
+        """
+        Sort or "shuffle" the underlying object.
+
+        "Shuffle" means the object is sorted so that all group members occur sequentially,
+        in the same chunk. Multiple groups may occur in the same chunk.
+        This method is particularly useful for chunked arrays (e.g. dask, cubed).
+        particularly when you need to map a function that requires all members of a group
+        to be present in a single chunk. For chunked array types, the order of appearance
+        is not guaranteed, but will depend on the input chunking.
```

Comment:
Is a single group limited to a single chunk? Assuming so, if we get one giant chuck, could that present any performance problems?

Reply:
The other chunks get "auto" reshaped. This is controlled by the `chunks` kwarg, which only takes "auto" at the moment. 

https://docs.dask.org/en/latest/generated/dask.array.shuffle.html